### PR TITLE
Improve Client CLI help readability

### PR DESCRIPTION
**Pull request by davxy**
Currently the CLI `-h/--help` commad output is almost unreadable as (for some commands) it:
- doesn't provide a short brief of what the command does.
- doesn't separate the options description in smaller paragraphs.
- doesn't use a smart wrap strategy for lines longer than the number of columns in the terminal.

Follow some pics taken with a 100 cols wide term

### Before

20231028-174531-grim

### After

20231028-175041-grim

### Before

20231028-175257-grim

### After

20231028-175155-grim

### CODE REVIEW

File: substrate/client/cli/src/params/node_key_params.rs [hunk 1]
```diff
@@ -32,39 +32,48 @@ const NODE_KEY_ED25519_FILE: &str = "secret_ed25519";
...
-	/// The secret key to use for libp2p networking.
+	/// Secret key to use for p2p networking.
+	///
 	/// The value is a string that is parsed according to the choice of
 	/// `--node-key-type` as follows:
-	///   `ed25519`:
-	///   The value is parsed as a hex-encoded Ed25519 32 byte secret key,
-	///   i.e. 64 hex characters.
+	///
+	///  - `ed25519`: the value is parsed as a hex-encoded Ed25519 32 byte secret key (64 hex
```
Comment (davxy) [inline substrate/client/cli/src/params/node_key_params.rs:40]:
```suggestion
	///  - `ed25519`: the value is parsed as a hex-encoded Ed25519 32 byte secret key (64 hex chars)
```

**Review by bkchr — APPROVED** [review]
The small things ❤️

**Comment by bkchr** [comment]
bot fmt

File: substrate/client/cli/src/params/import_params.rs [hunk 2]
```diff
@@ -115,26 +117,23 @@ impl ImportParams {
...
 	pub execution_block_construction: Option<ExecutionStrategy>,
 
-	/// The means of execution used when calling into the runtime while using an off-chain worker.
+	/// Runtime execution strategy for offchain workers.
 	#[arg(long, value_name = "STRATEGY", value_enum, ignore_case = true)]
 	pub execution_offchain_worker: Option<ExecutionStrategy>,
 
-	/// The means of execution used when calling into the runtime while not syncing, importing or
-	/// constructing blocks.
+	/// Runtime execution strategy for importing or constructing blocks while not syncing.
```
Comment (davxy) [inline substrate/client/cli/src/params/import_params.rs:136]:
```suggestion
	/// Runtime execution strategy when not syncing, importing or constructing blocks.
```

**Review by gupnik — APPROVED** [review]

**Comment by bkchr** [comment]
@davxy rustdoc test is still failing.

**Review by pepoviola — APPROVED** [review]

**Pull request merged**